### PR TITLE
modify runtime variable 'install.in_default' with rpm/deb

### LINKED ISSUE #3280: RPM and DEB agents goes to unhealthy state when Elastic Defend integration is added.
**Kibana Build details:**
```
VERSION: 8.10.0
BUILD: 66107
COMMIT: fa3473f42d7c5e7a3c2d66026a153e01002f5d3c
```

**Host OS and Browser version:** Windows, All

**Preconditions:**
1. 8.10.0 Kibana cloud environment should be available.
2. A Debian and RPM agent should be installed.

**Steps to reproduce:**
1. Navigate to Agents tab.
2. Navigates the DEB and RPM agent policy.
3. Add the `Elastic Defend` integration to the DEB and RPM agent policy.
4. After sometime, observe that RPM and DEB agents goes to unhealthy state when Elastic Defend integration is added.

**Agent.json:**
- Debian
[ip-172-31-75-216-agent-details.zip](https://github.com/elastic/elastic-agent/files/12415940/ip-172-31-75-216-agent-details.zip)
- RPM
[ip-172-31-65-98.ec2.internal-agent-details.zip](https://github.com/elastic/elastic-agent/files/12415955/ip-172-31-65-98.ec2.internal-agent-details.zip)

**Agent Logs:**
- Debian
[elastic-agent-diagnostics-2023-08-23T06-35-46Z-00.zip](https://github.com/elastic/elastic-agent/files/12415944/elastic-agent-diagnostics-2023-08-23T06-35-46Z-00.zip)

- RPM
[elastic-agent-diagnostics-2023-08-23T06-37-52Z-00.zip](https://github.com/elastic/elastic-agent/files/12415958/elastic-agent-diagnostics-2023-08-23T06-37-52Z-00.zip)

**Expected Result:**
RPM and DEB agents should not goes to unhealthy state when Elastic Defend integration is added.

**Note:**
- The issue is consistently reproducible for Debian and RPM agent.

**Screenshot:**

image (1)
image

### PR DESCRIPTION
**Pull request by leehinman**
## What does this PR do?

changes how the `install.in_default` runtime protection variable is calculated on systems where elastic-agent is installed via deb or rpm package.  Previously this would have returned false be cause the install path is not the same as the `tar.gz` install.  However, endpoint knows how to operate when elastic-agent is installed as a deb or rpm, so now it will return true.

## Why is it important?

This is required because the runtime prevention was blocking the install of endpoint on systems installed via deb/rpm.

example:

```
sudo elastic-agent status
┌─ fleet
│  └─ status: (HEALTHY) Connected
└─ elastic-agent
   ├─ status: (DEGRADED) 1 or more components/units in a failed state
   └─ endpoint-default
      ├─ status: (FAILED) Elastic Defend requires Elastic Agent be installed at the default installation path
      ├─ endpoint-default
      │  └─ status: (FAILED) Elastic Defend requires Elastic Agent be installed at the default installation path
      └─ endpoint-default-3575ea0b-e4a5-4b6d-b95e-a1ec04cb1683
         └─ status: (FAILED) Elastic Defend requires Elastic Agent be installed at the default installation path
```		 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
- [ ] I have added tests that prove my fix is effective or that my feature works
~~- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- [ ] I have added an integration test or an E2E test

## Author's Checklist

- [ ]

## How to test this PR locally

1. Debian or RPM system
2. Stack version 8.10.0
3. default version of elastic-agent 8.10.0
4. build custom version of elastic-agent off this PR, but change version to 8.10.0
5. create a policy with system and endpoint integrations
6. install default elastic-agent, should get error message above
7. uninstall elastic-agent
8. install custom elastic-agent, no error, agent should be healthy and endpoint sending data

## Related issues

- Closes #3280

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: internal/pkg/agent/install/pkgmgr/pkgmgr_unix.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+func InstalledViaExternalPkgMgr() bool {
+	if runtime.GOOS != "linux" {
+		return false
+	}
+
+	binaryName := paths.BinaryName
+
+	// NOTE searching for english words might not be a great idea as far as portability goes.
+	// list all installed packages then search for paths.BinaryName?
+	// dpkg is strange as the remove and purge processes leads to the package bing isted after a remove, but not after a purge
```
Comment (faec) [inline internal/pkg/agent/install/pkgmgr/pkgmgr_unix.go:28]: bing isted -> being listed? 😄

File: internal/pkg/agent/install/pkgmgr/pkgmgr_unix.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+	// check debian based systems (or systems that use dpkg)
+	// If the package has been installed, the status starts with "install"
+	// If the package has been removed (but not pruged) status starts with "deinstall"
```
Comment (faec) [inline internal/pkg/agent/install/pkgmgr/pkgmgr_unix.go:32]: purged

**Review by faec — APPROVED** [review]

**Review by cmacknz — APPROVED** [review]
That this code already exists and executes on every install makes me feel better about it not having tests yet.

**Comment by cmacknz** [comment]
No new failures, merging.

**Pull request merged**